Re-enable TestWebKitAPI.WebpagePreferences.WebsitePoliciesAutoplayQuirks on new macOS

achristensen07 opened the pull request:
#### 3857b6ae9b5240a9d2124501ac70c74b7d549cdf
<pre>
Re-enable TestWebKitAPI.WebpagePreferences.WebsitePoliciesAutoplayQuirks on new macOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=282214">https://bugs.webkit.org/show_bug.cgi?id=282214</a>
<a href="https://rdar.apple.com/137267112">rdar://137267112</a>

Reviewed by Wenson Hsieh.

Sometimes the TestWebKitAPI::Util::run call in TestWKWebView.waitForMessage causes
the runloop to spin until multiple messages have been received.  This makes it so
if you have multiple calls to waitForMessage without anything being done between them,
it can sometimes be waiting for a message it has already received, causing the test
to time out.  To fix this, use a Deque to be more robust against receiving multiple
messages in one call to Util::run.

A quick audit of the uses of TestWKWebView.waitForMessage found several more places
this is an issue, some of which had resulted in disabled tests.  I fixed this everywhere.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsitePolicies.mm:
(TEST(WebpagePreferences, WebsitePoliciesAutoplayQuirks)):
* Tools/TestWebKitAPI/cocoa/TestWKWebView.h:
* Tools/TestWebKitAPI/cocoa/TestWKWebView.mm:
(-[TestMessageHandler addMessage:withHandler:]):
(-[TestMessageHandler userContentController:didReceiveScriptMessage:]):
(-[TestWKWebView performAfterReceivingAnyMessage:]):
(-[TestMessageHandler setWildcardMessageHandler:]): Deleted.

Canonical link: <a href="https://commits.webkit.org/285843@main">https://commits.webkit.org/285843@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1864ffdfa8d1db3bac40cbba656148606157c980

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/73908 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/53337 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/26719 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/78244 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/25146 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/76025 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/62470 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/1122 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/58107 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/16463 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/76975 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/48263 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/63577 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/38506 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/45082 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/21066 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/23479 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/66631 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/21414 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/79798 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/1225 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/632 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/66438 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/1368 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/63590 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/65718 "Found 1 new API test failure: /TestWebKit:WebKit.InjectedBundleInitializationUserDataCallbackWins (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/9605 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/7783 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11416 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/1189 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/1218 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/1205 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/1224 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/1864ffdfa8d1db3bac40cbba656148606157c980. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/35837

achristensen07 committed:
Re-enable TestWebKitAPI.WebpagePreferences.WebsitePoliciesAutoplayQuirks on new macOS
https://bugs.webkit.org/show_bug.cgi?id=282214
rdar://137267112

Reviewed by Wenson Hsieh.

Sometimes the TestWebKitAPI::Util::run call in TestWKWebView.waitForMessage causes
the runloop to spin until multiple messages have been received.  This makes it so
if you have multiple calls to waitForMessage without anything being done between them,
it can sometimes be waiting for a message it has already received, causing the test
to time out.  To fix this, use a Deque to be more robust against receiving multiple
messages in one call to Util::run.

A quick audit of the uses of TestWKWebView.waitForMessage found several more places
this is an issue, some of which had resulted in disabled tests.  I fixed this everywhere.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsitePolicies.mm:
(TEST(WebpagePreferences, WebsitePoliciesAutoplayQuirks)):
* Tools/TestWebKitAPI/cocoa/TestWKWebView.h:
* Tools/TestWebKitAPI/cocoa/TestWKWebView.mm:
(-[TestMessageHandler addMessage:withHandler:]):
(-[TestMessageHandler userContentController:didReceiveScriptMessage:]):
(-[TestWKWebView performAfterReceivingAnyMessage:]):
(-[TestMessageHandler setWildcardMessageHandler:]): Deleted.

Canonical link: https://commits.webkit.org/285843@main
